src/core/schema.service.ts

```diff
@@ -38,20 +38,45 @@ const findAllRefs = (schema: JsonSchema, result: ReferenceSchemaMap = {}): Refer
 
   return result;
 };
-const addToArray = (key: string) => (data: Object) => (valueToAdd: object) => {
+const addToArray = (key: string) => (data: Object) => (valueToAdd: object, neighbourValue?: object,
+                                                       insertAfter = true) => {
   if (data[key] === undefined) {
     data[key] = [];
   }
   const childArray = data[key];
+  if (neighbourValue) {
+    const index = childArray.indexOf(neighbourValue);
+    if (insertAfter) {
+      if (index >= 0 && index < (childArray.length - 1)) {
+        childArray.splice(index + 1, 0, valueToAdd);
+
+        return;
+      }
+      // TODO proper logging
+      console.log('Warning: Could not add the new value after the given neighbour value.' +
+                  'The new value was added at the end.');
+    } else {
+      if (index >= 0) {
+        childArray.splice(index, 0, valueToAdd);
+
+        return;
+      }
+      // TODO proper logging
+      console.log('Warning: The given neighbour value could not be found.' +
+                  'The new value was added at the end.');
+    }
+  }
+  // default behavior: add at the end
   childArray.push(valueToAdd);
+
 };
 const deleteFromArray = (key: string) => (data: object) => (valueToDelete: object) => {
   const childArray = data[key];
   if (!childArray) {
     return;
   }
   const indexToDelete = childArray.indexOf(valueToDelete);
-  childArray.splice(indexToDelete - 1, 1);
+  childArray.splice(indexToDelete, 1);
```

Comment:
Makes sense, but it seems as we don't have any test case that reflects the incorrect delete behaviour (the commit does not alter any tests)? If so, could you add one?